src/System.Text.Json/src/System/Text/Json/Serialization/JsonPropertyInfoCommon.cs

```diff
@@ -69,19 +80,52 @@ internal override object GetValueAsObject(object obj, JsonSerializerOptions opti
                 return obj;
             }
 
-            Debug.Assert(Get != null);
+            Debug.Assert(HasGetter);
             return Get((TClass)obj);
         }
 
         internal override void SetValueAsObject(object obj, object value, JsonSerializerOptions options)
         {
-            Debug.Assert(Set != null);
+            Debug.Assert(HasSetter);
             TDeclaredProperty typedValue = (TDeclaredProperty)value;
 
             if (typedValue != null || !IgnoreNullPropertyValueOnWrite(options))
             {
-                Set((TClass)obj, (TDeclaredProperty)value);
+                Set(obj, typedValue);
             }
         }
+
+        internal TDeclaredProperty Get(object obj)
+        {
+            if (default(TClass) == null)
+            {
+                var get = (GetPropertyValue)_get;
+                return get((TClass)obj);
+            }
+            else
+            {
+                var get = (GetPropertyValueByRef)_get;
+                return get(ref Unsafe.As<ByReference>(obj).Value);
+            }
+        }
+
+        internal void Set(object obj, TDeclaredProperty value)
+        {
+            if (default(TClass) == null)
```

Comment:
Using `default` on every setter may be expensive. It seems like an easy thing to cache as a bool (class or struct).

Reply:
This is a well known optimization. The JIT eliminates the check and replaces it by a constant value (`true` for reference types and `Nullable<T>`, false for other value types).